### PR TITLE
Test fix SSLDriverTests.testHandshakeFailureBecauseProtocolMismatch

### PR DESCRIPTION
**Pull request by albertzaharovits**
The TLS protocol mismatch message during handshake in older JDK 8 is different. 

This is a fallout from https://github.com/elastic/elasticsearch/pull/67337

Closes https://github.com/elastic/elasticsearch/issues/67511

### CODE REVIEW

**Review by jkakavas — APPROVED** [review]
:shipit:

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/nio/SSLDriverTests.java [hunk 1]
```diff
@@ -163,21 +165,22 @@ public void testHandshakeFailureBecauseProtocolMismatch() throws Exception {
 
         final String[] serverProtocols;
         final String[] clientProtocols;
-        final String expectedMessage;
+        final Matcher expectedMessageMatcher;
```
Comment (Archish27) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/nio/SSLDriverTests.java:168]:
```suggestion
        private final Matcher expectedMessageMatcher;
```
Can we change it to private? Also, subsequent member variables.

Comment (albertzaharovits) [reply x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/nio/SSLDriverTests.java:168]: Thank you for the feedback, but this is a local variable which doesn't support access modifiers.

Comment (Archish27) [reply x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/nio/SSLDriverTests.java:168]: Ohh my bad 🙈

**Pull request merged**